src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -115,33 +115,112 @@ public static void Analyze(CSharpCompilation compilation, Symbol member, BoundNo
 
         protected override void Normalize(ref LocalState state)
         {
+            int oldNext = state.Capacity;
             state.EnsureCapacity(nextVariableSlot);
+            for (int slot = oldNext; slot < nextVariableSlot; slot++)
+            {
+                var value = GetDefaultState(ref state, slot);
+                state[slot] = value;
+            }
+        }
+
+        private bool? GetDefaultState(ref LocalState state, int slot)
+        {
+            if (slot == 0)
+            {
+                return null;
+            }
+
+            var variable = variableBySlot[slot];
+            var symbol = variable.Symbol;
+
+            switch (symbol.Kind)
+            {
+                case SymbolKind.Local:
+                    return null;
+                case SymbolKind.Parameter:
+                    {
+                        var parameter = (ParameterSymbol)symbol;
+                        return (parameter.RefKind == RefKind.Out) ?
+                            null :
```

Comment:
>null [](start = 28, length = 4)

The default state of an out parameter that hasn't been assigned is not null (true).

Reply:
Thanks. I'll include this in a subsequent PR that tracks unassigned values.